src/main.js

```diff
@@ -103,17 +122,29 @@ createNewBookButton.addEventListener('click', function(event) {
   descriptors.push(userDescriptor2.value)
 
   var userCreatedCover = buildNewCover(userCover.value, userTitle.value, userDescriptor1.value, userDescriptor2.value);
-  savedCovers.unshift(userCreatedCover);
-  console.log(userCreatedCover)
+
+  // savedCovers.unshift(userCreatedCover); //this could cause repeat covers if a user inputs the same cover data into the form twice
 
   updateCover(userCreatedCover);
+  saveVisibleCover();
 
   switchToHomeView();
 
   event.preventDefault();
 
 });
 
-function getRandomIndex(array) {
-  return Math.floor(Math.random() * array.length);
+// **** Save current cover functionality ****
+function saveVisibleCover() {
+  console.log(savedCovers);
+  var isRepeat = false;
+  for (var i = 0; i < savedCovers.length; i++) {
+    if (visibleCover === savedCovers[i]) {
+      isRepeat = true
+    }
+  }
+  if (isRepeat === false) {
+    savedCovers.unshift(visibleCover);
+  }
```

Comment:
Look into the array method `.includes()`. It returns a boolean if an array contains the search condition. You can use that to unshift your movie poster only if the includes method returns false. Utilize the bang `!` operator and you could accomplish what you're doing here really succinctly. Something like `if (!array.includes(thingToSearchFor) { doTheThing }`

This isn't WET code necessarily, but you're basically programming a built in array method, which is cool in and of itself.